data flows (#185)
AdobeDocs/commerce-extensibility: src/pages/app-development/starter-kit/customers.md

 - Tools
---

# `customer` data flow

This page describes the path that data takes as it travels between Adobe Commerce and your backoffice system when using the Adobe Commerce Extensibility Starter Kit.

> 

Reviewer: Should we include a diagram similar like this one in each one of the data flows? https://wiki.corp.adobe.com/display/EntComm/Product+Data+Object

> ## Create

Reviewer 2: ```suggestion
## Customer create
```

> Adobe Commerce origination

Reviewer 2: Global comment: Can we make this a full sentence?

Adobe Commerce supports the following ways to create a customer.

(I'm not happy wiith that suggestion, but I'm trying to start the sentence with AC and not use passive voice

> - A new customer registers in Adobe Commerce (sign up or request)

Reviewer 2: ```suggestion
- A shopper registers on the Create an Account page
```

> - A new customer registers in Adobe Commerce during or after purchase (sign up and purchase)

Reviewer 2: ```suggestion
- A shopper registers during or after purchase (sign up and purchase)
```

> - An admin user registers a new customer

Reviewer 2: Customers can also be created with REST and GraphQL. I think GraphQL would only be used to complete the UI actions, but REST might be used to create accounts in bulk

> Back office integration origination

Reviewer 2: Make this a sentence too

- A new or existing customer engages the company through sales, a partner, or a CRM, when records and emails are synched into Adobe Commerce.

> ## Update

Reviewer 2: ```suggestion
## Customer update
```

Adobe Commerce origination

- The customer edits information via My Account

> - An admin User modifies existing customer information (including assignment to a different customer group)

Reviewer 2: ```suggestion
- An admin user or process modifies existing customer information (including assignment to a different customer group)
```

By process I mean REST/GraphQL

Backoffice integrations support the following update methods:

- An existing customer engages the company through sales, a partner, or a CRM (including assignment to a different customer group)

> ## Create/Update

Reviewer 2: ```suggestion
## Customer group create/update
```
